src/machine/machined-varlink.c

```diff
@@ -424,16 +481,27 @@ typedef struct MachineLookupParameters {
 } MachineLookupParameters;
 
 static int vl_method_list(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        struct params {
+                const char *machine_name;
+                pid_t pid;
+                bool acquire_addresses;
+                bool acquire_os_release;
+                bool acquire_uid_shift;
+        };
+
         static const sd_json_dispatch_field dispatch_table[] = {
-                VARLINK_DISPATCH_MACHINE_LOOKUP_FIELDS(MachineLookupParameters),
+                VARLINK_DISPATCH_MACHINE_LOOKUP_FIELDS(struct params),
+                { "acquireAddresses", SD_JSON_VARIANT_BOOLEAN, sd_json_dispatch_stdbool, offsetof(struct params, acquire_addresses),  0 },
+                { "acquireOSRelease", SD_JSON_VARIANT_BOOLEAN, sd_json_dispatch_stdbool, offsetof(struct params, acquire_os_release), 0 },
+                { "acquireUIDShift",  SD_JSON_VARIANT_BOOLEAN, sd_json_dispatch_stdbool, offsetof(struct params, acquire_uid_shift),  0 },
```

Comment:
as mentioned, it might make sense to simplify this, and have a single knob to acquire all three.

but not entirely sure.

having three might be fine too. maybe it's even kinda nice if you can request some of these props "gracefully" and some of them shoudl fail